src/messages/embed.ts

```diff
@@ -0,0 +1,257 @@
+	 */
+	public addField(name: string, value: string, inline = false): this {
+		return this.addFields({ name, value, inline });
+	}
+
+	/**
+	 * Adds fields to the embed (max 25).
+	 * @param fields The fields to add
+	 */
+	public addFields(...fields: APIEmbedField[]): this {
+		this.fields.push(...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, and inserts fields in the embed (max 25).
+	 * @param index The index to start at
+	 * @param deleteCount The number of fields to remove
+	 * @param fields The replacing field objects
+	 */
+	public spliceFields(index: number, deleteCount: number, ...fields: APIEmbedField[]): this {
+		this.fields.splice(index, deleteCount, ...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Sets the author of this embed.
+	 * @param name The name of the author
+	 * @param iconURL The icon URL of the author
+	 * @param url The URL of the author
+	 */
+	public setAuthor(name: string, iconURL?: string, url?: string): this {
+		this.author = { name, icon_url: iconURL, url };
+		return this;
+	}
+
+	/**
+	 * Sets the color of this embed.
+	 * @param color The color of the embed
+	 */
+	public setColor(color: number): this {
+		this.color = color;
+		return this;
+	}
+
+	/**
+	 * Sets the description of this embed.
+	 * @param description The description
+	 */
+	public setDescription(description: string): this {
+		this.description = description;
+		return this;
+	}
+
+	/**
+	 * Sets the footer of this embed.
+	 * @param text The text of the footer
+	 * @param iconURL The icon URL of the footer
+	 */
+	public setFooter(text: string, iconURL: string): this {
+		this.footer = { text, icon_url: iconURL };
+		return this;
+	}
+
+	/**
+	 * Sets the image of this embed.
+	 * @param url The URL of the image
+	 */
+	public setImage(url: string): this {
+		this.image = { url };
+		return this;
+	}
+
+	/**
+	 * Sets the thumbnail of this embed.
+	 * @param url The URL of the thumbnail
+	 */
+	public setThumbnail(url: string): this {
+		this.thumbnail = { url };
+		return this;
+	}
+
+	/**
+	 * Sets the timestamp of this embed.
+	 * @param timestamp The timestamp or date
+	 */
+	public setTimestamp(timestamp: number | Date = Date.now()): this {
+		if (timestamp instanceof Date) this.timestamp = timestamp.toISOString();
+		else this.timestamp = new Date(timestamp).toISOString();
+		return this;
+	}
+
+	/**
+	 * Sets the title of this embed.
+	 * @param title The title
+	 */
+	public setTitle(title: string): this {
+		this.title = title;
+		return this;
+	}
+
+	/**
+	 * Sets the URL of this embed.
+	 * @param url The URL
+	 */
+	public setURL(url: string): this {
+		this.url = url;
+		return this;
+	}
+
+	/**
+	 * Transforms the embed to a plain object.
+	 */
+	public toJSON(): APIEmbed {
+		return {
+			title: this.title,
+			description: this.description,
+			url: this.url,
+			timestamp: this.timestamp ? new Date(this.timestamp).toISOString() : undefined,
```

Comment:
This is already an ISO string.
```suggestion
			timestamp: this.timestamp,
```